### PR TITLE
paid action payment methods as an array

### PR DESCRIPTION
**Pull request by huumn**
This takes generically useful stuff from the #1481 branch so we can use it ahead of time.

**This has not been tested yet**

### CODE REVIEW

File: api/paidAction/index.js [hunk 1]
```diff
@@ -30,9 +30,9 @@ export const paidActions = {
   DONATE
 }
 
-export default async function performPaidAction (actionType, args, context) {
+export default async function performPaidAction (actionType, args, incomingContext) {
```
Comment (riccardobl) [inline api/paidAction/index.js:33]: to make this prettier we could just use  { ...context }, like:
```
export default async function performPaidAction (actionType, args, { ...context }) {
```
on every function that requires the context, so they get a shallow copy each

[the remark this answers is not in the document]
Comment (huumn) [reply api/paidAction/index.js:33]: I didn't know you could do that! Will do!

[the remark this answers is not in the document]
Comment (huumn) [reply api/paidAction/index.js:33]: Actually I'm a bit burned out on testing this (it breaks in the most subtle ways if a field is missing) and I don't want to break anything lol.

I'll try to do it in a future PR when I need to test everything again anyway.

File: api/paidAction/zap.js [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
 export const anonable = true
-export const supportsPessimism = true
-export const supportsOptimism = true
+
+export const paymentMethods = [
+  PAID_ACTION_PAYMENT_METHODS.FEE_CREDIT,
+  PAID_ACTION_PAYMENT_METHODS.P2P,
+  PAID_ACTION_PAYMENT_METHODS.OPTIMISTIC,
+  PAID_ACTION_PAYMENT_METHODS.PESSIMISTIC
+]
```
Comment (huumn) [inline api/paidAction/zap.js:12]: When nov-5 changes go in `FEE_CREDIT` and `P2P` will just have to switch spots.

**Comment by huumn** [comment]
I merged this to unblock some of our related work and to get more incidental testing.

**Pull request merged**